### PR TITLE
samples: add update intent code sample

### PR DESCRIPTION
**Pull request by galdawave**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-dialogflow-cx/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

### CODE REVIEW

File: samples/snippets/src/main/java/dialogflow/cx/UpdateIntent.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2020 Google LLC
```
Comment (averikitsch) [inline samples/snippets/src/main/java/dialogflow/cx/UpdateIntent.java:2]:
```suggestion
 * Copyright 2021 Google LLC
```

File: samples/snippets/src/main/java/dialogflow/cx/UpdateIntent.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+import com.google.cloud.dialogflow.cx.v3.Intent;
+import com.google.cloud.dialogflow.cx.v3.Intent.Builder;
+import com.google.cloud.dialogflow.cx.v3.IntentsClient;
+import com.google.cloud.dialogflow.cx.v3.UpdateIntentRequest;
+import com.google.protobuf.FieldMask;
+import java.io.IOException;
+
+public class UpdateIntent {
+
```
Comment (averikitsch) [inline samples/snippets/src/main/java/dialogflow/cx/UpdateIntent.java:27]: Please add a main method https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md#method-structure

File: samples/snippets/src/main/java/dialogflow/cx/UpdateIntent.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    intentBuilder.setDisplayName(displayName);
+    FieldMask fieldMask = FieldMask.newBuilder().addPaths("display_name").build();
+
+    Intent intent = intentBuilder.build();
+    UpdateIntentRequest request = UpdateIntentRequest.newBuilder()
+        .setIntent(intent)
+        .setLanguageCode("en")
+        .setUpdateMask(fieldMask)
+        .build();
+    return client.updateIntent(request);
```
Comment (averikitsch) [inline samples/snippets/src/main/java/dialogflow/cx/UpdateIntent.java:47]: Samples usually return void to show what to with the response https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md#method-structure

File: samples/snippets/src/main/java/dialogflow/cx/UpdateIntent.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import com.google.cloud.dialogflow.cx.v3.IntentsClient;
+import com.google.cloud.dialogflow.cx.v3.UpdateIntentRequest;
+import com.google.protobuf.FieldMask;
+import java.io.IOException;
+
+public class UpdateIntent {
+
+  public static com.google.cloud.dialogflow.cx.v3.Intent updateIntent(String projectId,
+      String agentId, String intentId, String location, String displayName) throws IOException {
+    IntentsClient client = IntentsClient.create();
```
Comment (averikitsch) [inline samples/snippets/src/main/java/dialogflow/cx/UpdateIntent.java:30]: https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md#client-initialization

File: samples/snippets/src/main/java/dialogflow/cx/UpdateIntent.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+package dialogflow.cx;
+
+import com.google.cloud.dialogflow.cx.v3.Intent;
+import com.google.cloud.dialogflow.cx.v3.Intent.Builder;
+import com.google.cloud.dialogflow.cx.v3.IntentsClient;
+import com.google.cloud.dialogflow.cx.v3.UpdateIntentRequest;
+import com.google.protobuf.FieldMask;
+import java.io.IOException;
+
+public class UpdateIntent {
```
Comment (averikitsch) [inline samples/snippets/src/main/java/dialogflow/cx/UpdateIntent.java:26]: Please see https://g3doc.corp.google.com/company/teams/cloud-devrel/dpe/samples/snippets.md?cl=head#rubric for adding appropriate comments

**Review by averikitsch — CHANGES_REQUESTED** [review]

**Review by galdawave — COMMENTED** [review]
Resolved the comments

File: samples/snippets/src/main/java/dialogflow/cx/UpdateIntent.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+      Intent intent = intentBuilder.build();
+      UpdateIntentRequest request = UpdateIntentRequest.newBuilder()
+          .setIntent(intent)
+          .setLanguageCode("en")
+          .setUpdateMask(fieldMask)
+          .build();
+    
+
+      //Make API request to update intent using fieldmask
+      Intent response = client.updateIntent(request);
```
Comment (averikitsch) [inline samples/snippets/src/main/java/dialogflow/cx/UpdateIntent.java:62]: Is there anything you'd like you highlight in this response? Is the print statement show useful info?

Comment (galz10) [reply samples/snippets/src/main/java/dialogflow/cx/UpdateIntent.java:62]: There's nothing really useful in the response it just returns the inputs you provided and the agent location

**Review by averikitsch — APPROVED** [review]

**Review by Neenu1995 — APPROVED** [review]

**Pull request merged**